lib/ddtrace/contrib/redis/patcher.rb

```diff
@@ -72,16 +73,28 @@ def call_pipeline(*args, &block)
             def datadog_pin
               @datadog_pin ||= begin
                 pin = Datadog::Pin.new(
-                  Datadog.configuration[:redis][:service_name],
+                  datadog_configuration[:service_name],
                   app: Ext::APP,
                   app_type: Datadog::Ext::AppTypes::DB,
                   tracer: Datadog.configuration[:redis][:tracer]
                 )
                 pin.onto(self)
               end
             end
+
+            private
+
+            def datadog_configuration
+              @datadog_configuration ||= Datadog.configuration[:redis, resolver.resolve(options)]
+            end
+
+            def resolver
```

Comment:
In keeping with the comment above, I would recommend removing this method.

Reply:
I understand your point, but it is not possible to retrieve exactly the same configuration without resolving the client instance options:
* some of the options may vary after that an instance connects for the first time (e.g. the `url` is built out of the configuration hash at the connection time, so the options passed to the `describes` key will never match a client instance)
* we discussed about taking a subset of keys according to the connection scheme (unix vs redis | rediss)

I will still need to normalize the connection option to match the configuration of the APM

(I can extract the function in another module and import it in the client eventually)
```rb
          def normalize(hash)
            return { url: hash[:url] } if hash[:scheme] == 'unix'

            # Connexion strings are always converted to host, port, db and scheme
            # but the host, port, db and scheme will generate the :url only after
            # establishing a first connexion
            {
              host: hash[:host],
              port: hash[:port],
              db: hash[:db],
              scheme: hash[:scheme]
            }
          end
```

Reply:
Now I got it.

When you call `Datadog.configuration[:redis, {}]` it will automatically go through the resolver.
This is something I missed 😄